File: kotlin-ide/idea/testData/refactoring/pullUp/j2k/moveSuperInterfacesWithGenerics.java
// INFO: {"checked": "true"}
interface X<T, U> {

}

// INFO: {"checked": "false"}
interface Y<V> {

}

// INFO: {"checked": "true"}
interface Z<T> {

}

class P {

}

class <caret>B<S, U, V> extends A<S> implements X<P, S>, Y<U>, Z<V> {

}

File: kotlin-ide/idea/testData/refactoring/pullUp/k2j/moveSuperInterfacesWithGenerics.kt
// INFO: {"checked": "true"}
interface X<T, U>

// INFO: {"checked": "true"}
interface Y<V : Any>

// INFO: {"checked": "true"}
interface Z<T>

class <caret>B<S : Any, U : S, V>: A<S>(), X<U, V>, Y<S>, Z<U>